src/NetAnalyzers/Microsoft.CodeAnalysis.NetAnalyzers.md

```diff
@@ -1740,6 +1740,42 @@ Prefer a 'TryAdd' call over an 'Add' call guarded by a 'ContainsKey' check. 'Try
 |CodeFix|True|
 ---
 
+## [CA1865](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1865): Use char overload
+
+The char overload is a better performing overload for a string with a single char.
+
+|Item|Value|
+|-|-|
+|Category|Performance|
+|Enabled|True|
+|Severity|Info|
+|CodeFix|True|
+---
+
+## [CA1866](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1866): Use char overload
+
+The char overload is a better performing overload for a string with a single char.
+
+|Item|Value|
+|-|-|
+|Category|Performance|
+|Enabled|True|
+|Severity|Info|
+|CodeFix|False|
+---
+
+## [CA1867](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1867): Use char overload
+
+The char overload is a better performing overload for a string with a single char.
+
+|Item|Value|
+|-|-|
+|Category|Performance|
+|Enabled|False|
+|Severity|Warning|
```

Comment:
>Warning| [](http://example.com/codeflow?start=10&length=8)

I know this diagnostic is not enabled by default but why we need to having it as warning if anyone enable it? maybe setting this as Info will be consistent with the others.

Reply:
Not sure why this was generated as Warning. It's `RuleLevel.Disabled` in the code:
https://github.com/dotnet/roslyn-analyzers/blob/26432129d87f12e8b5a07d3f5173e13594338766/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseStringMethodCharOverloadWithSingleCharacters.cs#L43-L51

Should I use some other value? (I'm guessing I can't edit this directly in the Microsoft.CodeAnalysis.NetAnalyzers.md file since this was updated by running the pack command.)

Reply:
Looking at CA1001 which is being displayed as "Hidden" here, it seems to be using `RuleLevel.IdeHidden_BulkConfigurable`. Is that what we want here?

Reply:
But then again, looking at all the other rules in this file that have `Enabled|False`, they all seem to have a Warning severity. So maybe this is fine?

Reply:
@buyaa-n do you have suggestion here? I mean to have set the rule as disabled and if someone enable it be "Info" and not `Warining`?

Reply:
I think it was intentional that rules that are disabled by default are warnings.

Reply:
I care more about the behavior when enabling this specific rule. It is better to be `Info` when it is enabled.

Reply:
> I know this diagnostic is not enabled by default but why we need to having it as warning if anyone enable it

Probably it doesn't matter as disabled rules are not bulk configurable, they need to enabled with their ID specifically, for that the user need to specify the severity they want which would override this warning level, example:

`dotnet_diagnostic.CA1867.severity = info`

Reply:
> Looking at CA1001 which is being displayed as "Hidden" here, it seems to be using `RuleLevel.IdeHidden_BulkConfigurable`. Is that what we want for "Severity = none"? (would have thought `RuleLevel.Disabled` is the right one for severity none)

Probably better to ask in the issue so that API approvers could answer

Reply:
I am fine then with what we have now.